python/pyarrow/types.pxi

```diff
@@ -591,6 +591,45 @@ cdef class Schema:
         return self.sp_schema.get().Equals(deref(_other.schema),
                                            check_metadata)
 
+    @classmethod
+    def from_pandas(cls, df, bint preserve_index=True):
+        """
+        Returns implied schema from dataframe
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+        preserve_index : bool, optional
+            Whether to store the index as an additional column in the resulting
+            ``Table``.
+
+        Returns
+        -------
+        pyarrow.Schema
+
+        Examples
+        --------
+
+        >>> import pandas as pd
+        >>> import pyarrow as pa
+        >>> df = pd.DataFrame({
+            ...     'int': [1, 2],
+            ...     'str': ['a', 'b']
+            ... })
+        >>> pa.Schema.from_pandas(df)
+        int: int64
+        str: string
+        __index_level_0__: int64
```

Comment:
Do you want to add an example of this to the Sphinx documentation?

Reply:
pandas.rst lacks of examples, so just mentioned it :)